thriftrw/spec/common.pyx

```diff
@@ -34,6 +34,24 @@ def fields_eq(fields):
     return __eq__
 
 
+def fields_hash(fields):
+    """Generates an ``__hash__`` method.
+
+    :param list fields:
+        List of fields of the object to be hashed.
+    """
+    def hashable(obj):
+        try:
+            hash(obj)
+        except TypeError:
+            return False
+        return True
+
+    def __hash__(self):
```

Comment:
We should probably skip `__hash__` generation for structs and unions if all
their fields aren't hashable.

Otherwise a struct with lists/maps/sets only shows as hashable even though
it's basically always returning `hash(())`.